File: C++/LabWork#5/Symbols.cpp
#include "Symbols.h"

Symbols::Symbols(std::string val) : Row(val) {}

void Symbols::reduce() {
  std::string temp = "";
  for (int i = 0; i < value.size(); i++) {
    if (i % 2 == 0)
      temp += value[i];
  }
  value = temp;
}

File: C#/LabWork#5/Symbols.cs
using System;


namespace LabWork_5
{
  public class Symbols : Row
  {
    public Symbols(string val) : base(val) { }

    public override void Reduce()
    {
      string temp = "";
      for (int i = 0; i < value.Length; i++)
      {
        if (i % 2 == 0)
          temp += value[i];
      }
      value = temp;
    }
  }


}
